### PR TITLE
[PF-811] Clone Referenced Resources

### PR DESCRIPTION
**Pull request by jaycarlton**
Create clones of all reference types that point to the original referent. This didn't involve any new flights, but reused the create reference flight.

### CODE REVIEW

File: service/src/main/resources/api/service_openapi.yaml [hunk 1]
```diff
@@ -1965,7 +2010,26 @@ components:
...
+        sourceWorkspaceId:
+          description: ID of the workspace of the source resource for this clone
+          type: string
+          format: uuid
+        sourceResourceId:
+          description: ID of the source resource
+          type: string
+          format: uuid
+        resource:
+          $ref: '#/components/schemas/ResourceDescription'
```
Comment (ddietterich) [inline service/src/main/resources/api/service_openapi.yaml:2032]: I have mixed feelings about this. I would like to minimize the use of the union form. It is annoying to clients, as we have heard. I think we could put the clone operation at the per-resource level with a specific response and avoid it.

**Review by ddietterich — COMMENTED** [review]
Very nice. I would like to talk about the API.

I think I would prefer to have per-resource endpoints as we do with everything except enumeration.

File: service/src/main/resources/api/service_openapi.yaml (hunk 1 above)
Comment (jaycarlton) [reply service/src/main/resources/api/service_openapi.yaml:2032]: I'll split them out; it won't be too bad.

**Review by ddietterich — APPROVED** [review]
LGTM

File: integration/src/main/java/scripts/testscripts/CloneReferencedResources.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            .id(destinationWorkspaceId)
+            .spendProfile(getSpendProfileId())
+            .stage(getStageModel());
+    final CreatedWorkspace createdDestinationWorkspace = workspaceApi.createWorkspace(requestBody);
+    assertThat(createdDestinationWorkspace.getId(), equalTo(destinationWorkspaceId));
+
+    // create destination cloud context
+    String destinationProjectId = CloudContextMaker
+        .createGcpCloudContext(destinationWorkspaceId, workspaceApi);
+    logger.info("Created destination project {} in workspace {}", destinationProjectId, destinationWorkspaceId);
```
Comment (zloery) [inline integration/src/main/java/scripts/testscripts/CloneReferencedResources.java:83]: If we're just cloning references, I don't think we need a cloud context.

**Review by zloery — APPROVED** [review]
one small q in the test but LGTM otherwise!

File: integration/src/main/java/scripts/testscripts/CloneReferencedResources.java (hunk 2 above)
Comment (jaycarlton) [reply integration/src/main/java/scripts/testscripts/CloneReferencedResources.java]: Awesome! Removing that takes the test time down to 30 seconds.

**Pull request merged**